src/intro-to-c/conditionals/conditionals.md

```diff
@@ -0,0 +1,284 @@
+
+``` c
+condition ? expression_1 : expression_2;
+```
+
+This is a rewrite of this equivalent `if-else` statement:
+
+``` c
+if(condition)
+{
+	expression_1;
+}
+else
+{
+	expression_2
+}
+```
+
+Lets first look at it with our blinds example:
+
+``` c
+#include <stdio.h>
+
+int main(void)
+{
+	int sunny = is_it_sunny();
+
+	(sunny == 1) ? close_the_blinds() : keep_the_blinds_open();
+
+	return 0;
+}
+```
+
+Here is a different example with a condition that compares two integers:
+
+``` c
+#include <stdio.h>
+
+int main(void)
+{
+	int num1 = 20, num2 = 23;
+
+	(num1 > num2) ? printf("%d is greater than %d\n", num1, num2) : printf("%d is greater than %d\n", num2, num1);
+
+	return 0;
+}
+```
+
+## Multi-condition Statements
+
+---
+
+`if` statements and the ternary operator can actually handle multiple comparisons or conditions. Looking back at the example in the `else if` section, it can be more efficiently rewritten like this:
+
+``` c
+#include <stdio.h>
+
+int main(void)
+{
+	if(sunny == 1 || dark == 1)
+	{
+		close_the_blinds();
+	}
+	else if(snowy == 1 || rainy == 1)
+	{
+		keep_the_blinds_open();
+	}
+	else
+	{
+		go_for_a_drive();
+	}
+	return 0;
+}
+```
+
+Let's try it with the ternary operator.
+
+``` c
+#include <stdio.h>
+
+int main(void)
+{
+	int m = 20, n = 15, g = 17, p = 3;
+
+	(m > n && m > g && m > p) ? printf("%d is the greatest.\n", m) : printf("%d is not the greatest.\n", m);
+
+return 0;
+}
+```
+
+Order doesn't technically matter, but the best practice is to order them in a way that allows for potential short circuiting.
+
+## Short Circuiting
+
+---
+
+When writing conditional statements, it is good practice to write them in a way that allows for the compiler to skip the evaluation of further conditions when possible. This skip by the compiler is called short circuiting. Short circuiting is possible when evaluating logical `and` and logical `or`. For example, if the left side of a logical and is false, the whole expression will be false regardless of the right side of the expression, so the compiler skips this evaluation. Here is an example of such a case:
+
+``` c
+#include <stdio.h>
+
+int main(void)
+{
+	int x = 1, y = 2, z = 2;
+
+	if(x == y && y == z)
+	{
+		printf("Everything is the same!\n");
+	}
+	else
+	{
+		printf("Everything is NOT the same!\n");
+	}
+	return 0;
+}
+```
+
+`x` and `y` are not equivalent, so the comparison would resolve to false, rendering the whole statement false.
+
+With a slight change, here is a similar example for a logical or short circuit:
+
+``` c
+#include <stdio.h>
+
+int main(void)
+{
+	int x = 1, y = 2, z = 2;
+
+	if(y == z || x == y)
+	{
+		printf("Something is the same!\n");
+	}
+	else
+	{
+		printf("Nothing is the same!\n");
+	}
+	return 0;
+}
+```
+
+Note that logical or will resolved if **either** side of the expression is true, so comparing `y` and `z` will resolve to true, rendering the whole statement true.
```

Comment:
I would add an extra sentence or two here just to flesh out this wording. Felt like it was just missing something.
```suggestion
Note that logical or will resolved if **either** side of the expression is true, so comparing `y` and `z` will resolve to true, rendering the whole statement true. Since `y == z` is true, `x` and `y` don't even need to be compared. The result of `x == y` would have no effect on the truthiness of the logical expression.
```